### PR TITLE
Features/duplicatedetectionrules

### PR DESCRIPTION
**Pull request by GabrielFG**
- Now exporting duplicate detection rules to bring across environments
- Rules need to be unpublished prior to exportation, this means:
    1. the powershell script in this PR will upublish/publish the rules for you
    2. The "Export and Commit Solution Data" pipeline will use the powershell script to unpublish, export and then re-publish the rules in the given environment when done.
- Deleting certain default rules when importing, this will only be needed when first creating a new environment and importing the package for the first time. The rules then won't come back and have no way to be accidently be exported into our data.

### CODE REVIEW

File: src/Compliance.Solution/Data/Lookups/data.xml [hunk 1]
```diff
@@ -8,15 +8,15 @@
         <field name="opc_name" value="Privacy Act" />
         <field name="statecode" value="0" />
         <field name="statuscode" value="1" />
-        <field name="ownerid" value="e3f1c462-47fa-4e41-8b27-ff84842afe3b" lookupentity="systemuser" lookupentityname="# SVC_OPC_CRM_Admin" />
+        <field name="ownerid" value="a5d21787-0ed1-ea11-a813-000d3a0c5524" lookupentity="systemuser" lookupentityname="Package Deployer" />
```
Comment (pierluc-ssoucy) [inline src/Compliance.Solution/Data/Lookups/data.xml:11]: It seems like are overriding the owner of all the records, not sure if that is what we should be doing.

File: azure-pipelines/unpack-commit-solution-data.yml [hunk 2]
```diff
@@ -32,6 +39,14 @@ steps:
     schemaFile: src/$(SolutionName).Solution/lookups-schema.xml
     dataFile: '$(Agent.TempDirectory)/data.zip'
 
+- task: PowerShell@2
+  displayName: 'Republish all duplicate rules to set them to their original state'
+  inputs:
+    filePath: 'src/$(SolutionName).Package/set_duplicaterule_publishstate.ps1'
+    arguments: '-ConnectionString "AuthType=ClientSecret;Url=$(crm.url);ClientId=$(tenant.appId);ClientSecret=$(tenant.secret)" -PublishState "published"'
+    workingDirectory: 'src/$(SolutionName).Package'
+  condition: always()
```
Comment (pierluc-ssoucy) [inline azure-pipelines/unpack-commit-solution-data.yml:48]: Why is this needed?

File: src/Compliance.Package/set_duplicaterule_publishstate.ps1 [hunk 3]
```diff
@@ -0,0 +1,46 @@
+﻿Param(
```
Comment (pierluc-ssoucy) [inline src/Compliance.Package/set_duplicaterule_publishstate.ps1:1]: Unless those scripts are used by the Compliance.Package project, I believe this should be at the same spot as the other pipeline scripts.

https://github.com/opc-cpvp/OPC.PowerApps/tree/master/azure-pipelines/scripts

**Review by pierluc-ssoucy — CHANGES_REQUESTED** [review]

File: azure-pipelines/unpack-commit-solution-data.yml (hunk 2 above)
Comment (GabrielFG) [reply azure-pipelines/unpack-commit-solution-data.yml:48]: @pierluc-ssoucy As the rules are unpublished, they then need to be re-published to be working in the environment again.

File: src/Compliance.Package/set_duplicaterule_publishstate.ps1 (hunk 3 above)
Comment (GabrielFG) [reply src/Compliance.Package/set_duplicaterule_publishstate.ps1:1]: Didn't know that existed. Will do.

File: src/Compliance.Solution/Data/Lookups/data.xml (hunk 1 above)
Comment (GabrielFG) [reply src/Compliance.Solution/Data/Lookups/data.xml:11]: I'm not sure how that happened, must be when publishing the package through the pipeline. I'll see what I can do to revert this.

File: azure-pipelines/unpack-commit-solution-data.yml (hunk 2 above)
Comment (GabrielFG) [reply azure-pipelines/unpack-commit-solution-data.yml:48]: or do you mean the condition? The condition makes sure the duplicate rules will get republished even if the the package import fails.

File: src/Compliance.Solution/Data/Lookups/data.xml [hunk 4]
```diff
@@ -1464,24 +1464,7 @@
...
-          <targetid>a2d81b68-b53e-ea11-a812-000d3af464f8</targetid>
-        </targetids>
-      </m2mrelationship>
-      <m2mrelationship sourceid="4645d1a6-8d5c-ea11-a811-000d3af46fff" targetentityname="team" targetentitynameidfield="teamid" m2mrelationshipname="opc_queues_teams">
-        <targetids>
-          <targetid>9ed81b68-b53e-ea11-a812-000d3af464f8</targetid>
-        </targetids>
-      </m2mrelationship>
-    </m2mrelationships>
+    <m2mrelationships />
```
Comment (GabrielFG) [inline src/Compliance.Solution/Data/Lookups/data.xml:1467]: @pierluc-ssoucy  Some missing relationships here. Is this normal?

File: src/Compliance.Solution/Data/Lookups/data.xml (hunk 1 above)
Comment (GabrielFG) [reply src/Compliance.Solution/Data/Lookups/data.xml]: Changed them back with a search and replace, still not sure the reasoning behind this owner change though.

File: src/Compliance.Solution/Data/Lookups/data.xml (hunk 4 above)
Comment (pierluc-ssoucy) [reply src/Compliance.Solution/Data/Lookups/data.xml:1467]: good catch, i missed this. The schema seems fine though, I guess you don't have the data for some reason

Comment (joshcampbell191) [reply src/Compliance.Solution/Data/Lookups/data.xml:1467]: Try resetting your environment to see if that fixes the problem.

**Review by pierluc-ssoucy — APPROVED** [review]

**Pull request merged**